pkg/client/apiutil/apimachinery.go

```diff
@@ -64,3 +66,30 @@ func RESTClientForGVK(gvk schema.GroupVersionKind, baseConfig *rest.Config, code
 	}
 	return rest.RESTClientFor(cfg)
 }
+
+// RESTUnstructuredClientForGVK constructs a new rest.Interface for accessing unstructured resources.
+func RESTUnstructuredClientForGVK(gvk schema.GroupVersionKind, baseConfig *rest.Config) (rest.Interface, error) {
+	gv := gvk.GroupVersion()
+
+	cfg := rest.CopyConfig(baseConfig)
+	cfg.GroupVersion = &gv
+	if gvk.Group == "" {
+		cfg.APIPath = "/api"
+	} else {
+		cfg.APIPath = "/apis"
+	}
+	var jsonInfo runtime.SerializerInfo
+	for _, info := range scheme.Codecs.SupportedMediaTypes() {
+		if info.MediaType == runtime.ContentTypeJSON {
+			jsonInfo = info
+			break
+		}
+	}
+	jsonInfo.Serializer = unstructured.UnstructuredJSONScheme
+
+	cfg.NegotiatedSerializer = serializer.NegotiatedSerializerWrapper(jsonInfo)
```

Comment:
the only thing that changes here is the negotiated serializer setup, right?  Refactor the common bits out into a private function so we're not copying and pasting code around.

Reply:
Also, do you have to care about status, like the old dynamic client did?